### PR TITLE
Upgrade to AndroidX

### LINKED ISSUE #30: AndroidX Conversion
Hey been using your library for awhile, but wanted to bring this to your attention. Looks like soon they are going to be removing non-AndroidX support. Is this something you have looked into? If you do not have time to do it, I may take a crack and converting it.

Output from Gradle currently:

_Your project has set `android.useAndroidX=true`, but configuration `:app:debugRuntimeClasspath` still contains legacy support libraries, which may cause runtime issues.
This behavior will not be allowed in Android Gradle plugin 8.0.
Please use only AndroidX dependencies or set `android.enableJetifier=true` in the `gradle.properties` file to migrate your project to AndroidX (see https://developer.android.com/jetpack/androidx/migrate for more info).
The following legacy support libraries are detected:_
```
:app:debugRuntimeClasspath -> com.github.AliAsadi:PowerPreference:2.1.0 -> com.android.support:recyclerview-v7:25.0.0
:app:debugRuntimeClasspath -> com.github.AliAsadi:PowerPreference:2.1.0 -> com.android.support:recyclerview-v7:25.0.0 -> com.android.support:support-annotations:25.0.0
:app:debugRuntimeClasspath -> com.github.AliAsadi:PowerPreference:2.1.0 -> com.android.support:recyclerview-v7:25.0.0 -> com.android.support:support-compat:25.0.0
:app:debugRuntimeClasspath -> com.github.AliAsadi:PowerPreference:2.1.0 -> com.android.support:recyclerview-v7:25.0.0 -> com.android.support:support-core-ui:25.0.0
:app:debugRuntimeClasspath -> com.github.AliAsadi:PowerPreference:2.1.0 -> com.android.support:appcompat-v7:25.0.0
:app:debugRuntimeClasspath -> com.github.AliAsadi:PowerPreference:2.1.0 -> com.android.support:appcompat-v7:25.0.0 -> com.android.support:support-v4:25.0.0
:app:debugRuntimeClasspath -> com.github.AliAsadi:PowerPreference:2.1.0 -> com.android.support:appcompat-v7:25.0.0 -> com.android.support:support-v4:25.0.0 -> com.android.support:support-media-compat:25.0.0
:app:debugRuntimeClasspath -> com.github.AliAsadi:PowerPreference:2.1.0 -> com.android.support:appcompat-v7:25.0.0 -> com.android.support:support-v4:25.0.0 -> com.android.support:support-core-utils:25.0.0
:app:debugRuntimeClasspath -> com.github.AliAsadi:PowerPreference:2.1.0 -> com.android.support:appcompat-v7:25.0.0 -> com.android.support:support-v4:25.0.0 -> com.android.support:support-fragment:25.0.0
:app:debugRuntimeClasspath -> com.github.AliAsadi:PowerPreference:2.1.0 -> com.android.support:appcompat-v7:25.0.0 -> com.android.support:support-vector-drawable:25.0.0
:app:debugRuntimeClasspath -> com.github.AliAsadi:PowerPreference:2.1.0 -> com.android.support:appcompat-v7:25.0.0 -> com.android.support:animated-vector-drawable:25.0.0

```

### PR DESCRIPTION
**Pull request by deadman96385**
Fixes #30

I used the automated converter, which worked overall other then I had to also get an updated copy of Expandable Recycler. This was needed because AndroidX changes the Recyclerview some and so the support library you use needed updates. I've tested this against both the your sample app as well as the expandable recyclerview sample app. I haven't noticed any breakages.

https://github.com/deadman96385/expandable-recycler-view/commits/master

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -8,9 +8,9 @@ buildscript {
     ]
 
     ext.deps = [
-            appCompat             : "com.android.support:appcompat-v7:25.0.0",
-            recyclerView          : "com.android.support:recyclerview-v7:25.0.0",
-            expandablerecyclerview: "com.github.thoughtbot.expandable-recycler-view:expandablerecyclerview:v1.3",
+            appCompat             : 'androidx.appcompat:appcompat:1.0.0',
+            recyclerView          : 'androidx.recyclerview:recyclerview:1.0.0',
+            expandablerecyclerview: 'com.github.deadman96385:expandable-recycler-view:v1.8',
```
Comment (AliAsadi) [inline build.gradle:13]: Regarding the expandablerecyclerview don't you think relying on your expandablerecyclerview implementation might be a little risky? If you removed it, there might be a problem. Perhaps consider a different option, such as copying the ARR into the library would be a good idea. What do you think?

Comment (AliAsadi) [inline build.gradle]: Regarding the expandablerecyclerview don't you think relying on your expandablerecyclerview implementation might be a little risky? If you removed it, there might be a problem. Perhaps consider a different option, such as copying the ARR into the library would be a good idea. What do you think?

File: power-preference/build.gradle [hunk 2]
```diff
@@ -14,6 +14,7 @@ android {
             minifyEnabled false
         }
     }
+    namespace 'com.preference'
```
Comment (AliAsadi) [inline power-preference/build.gradle:17]: I was unfamiliar with the namespace, nice!

I'm curious as to why it is preferred over the placing it in the AndroidManifest?

**Review by AliAsadi — COMMENTED** [review]
Looks great! Thank you for making those changes. 🙌🏼

Left some minor questions, let me know what do you think.

File: build.gradle (hunk 1 above)
Comment (deadman96385) [reply build.gradle]: Yeah that's an option, I wasn't sure what you wanted for to do, so I matched the existing. If you want you could fork it and we point at that instead. Then you would have final say, and if it needs future updates I am more then willing to pull request there.

File: power-preference/build.gradle (hunk 2 above)
Comment (deadman96385) [reply power-preference/build.gradle:17]: I detailed it in the AndroidX update commit but here it is again for reference

Removed the applicationid from the manifest and converted the library and sample app to use Namespace as needed by newer gradle. Application id is recommended to match namespace. Which you can see below:

https://developer.android.com/studio/build/configure-app-module#set-namespace:~:text=application%20ID.-,Change%20the%20namespace,-In%20most%20cases

File: build.gradle (hunk 1 above)
Comment (AliAsadi) [reply build.gradle]: Forked 👍🏼. Can you please update the implementation to use mine? Thank you.

```
implementation 'com.github.AliAsadi:expandable-recycler-view:v1.8'
```

Comment (deadman96385) [reply build.gradle]: Should be good to go now, I validated my branch in android studio with building and testing the sample app.

**Review by AliAsadi — APPROVED** [review]
💯

**Pull request merged**